src/Field.tsx

```diff
@@ -92,17 +102,32 @@ export function Field({
   } = useFormikContext();
 
   warning(
-    render,
-    '<Field render> has been deprecated and will be removed in future versions of Formik. Please use a function as a child instead.'
+    !!render,
+    `<Field render> has been deprecated and will be removed in future versions of Formik. Please use a child callback function instead. To get rid of this warning, 
+        replace 
+          <Field name="${name}" render={({field, form}) => ...} />
+        with
+          <Field name="${name}">{({field, form}) => ...}</Field>
+    `
+  );
+
+  warning(
+    !!component,
+    '<Field component> has been deprecated and will be removed in future versions of Formik. Use <Formik as> instead. Note that with the `as` prop, all props are passed directly through and not grouped in `field` object key.'
   );
 
   warning(
-    component && children && isFunction(children),
-    'You should not use <Field component> and <Field children> as a function in the same <Field> component; <Field component> will be ignored.'
+    !!is && !!children && isFunction(children),
```

Comment:
Since line 94 defaulted https://github.com/jaredpalmer/formik/pull/1406/files#diff-bc78961df770c0df68f6cd758a6f2afdR94 `as` to `input`, wouldn't this error message show up every time the dev does this?

```jsx
<Field name="name">
{({form, field}) => {
    return <MyCustomInput {...field} />
}}
</Field>
```

In the example above, as would default to `input`. In this case, would the check be something like `as !== 'input'`?

Reply:
@jaredpalmer  Why we Field render will be deprecated. 
Recently, we started using Field render in project. Formik 1.5.8
Please update API examples, if formik going to change ?